oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/method/annotation/OAuth2AuthorizedClientArgumentResolver.java

```diff
@@ -172,6 +191,24 @@ private OAuth2AuthorizedClient authorizeClientCredentialsClient(ClientRegistrati
 		return authorizedClient;
 	}
 
+	private boolean shouldRefreshToken(OAuth2AuthorizedClient authorizedClient) {
+		if (this.authorizedClientRepository == null) {
+			return false;
+		}
+		OAuth2RefreshToken refreshToken = authorizedClient.getRefreshToken();
+		if (refreshToken == null) {
+			return false;
+		}
+		return hasTokenExpired(authorizedClient);
+	}
+
+	private boolean hasTokenExpired(OAuth2AuthorizedClient authorizedClient) {
+		Instant now = this.clock.instant();
+		Instant expiresAt = authorizedClient.getAccessToken().getExpiresAt();
+
+		return now.isAfter(expiresAt.minus(this.accessTokenExpiresSkew));
```

Comment:
In OAuth2 token response, `expires_in` is an optional response data so this value can be `null`. 

In such scenarios, I am not sure if this is determined to be a JWT access token and has `exp` claim, then can it be leveraged to make this check.